src/gt4py/analysis/passes.py

```diff
@@ -1210,6 +1210,95 @@ def apply(self, transform_data: TransformData) -> TransformData:
         return transform_data
 
 
+class ReduceTemporaryStoragesPass(TransformPass):
+    class StorageReducer(gt_ir.IRNodeMapper):
```

Comment:
IRNodeMapper would be used to change the structure of the tree, e.g. when pruning nodes. since you modify all nodes in place, IRNodeVisitor seems more appropriate.

Reply:
(This is similar to ast.NodeVisitor and ast.NodeTransformer https://python.readthedocs.io/en/latest/library/ast.html#ast.NodeVisitor)